app/assets/javascripts/application.js

```diff
@@ -1,107 +1,124 @@
-    var title = links[i].getAttribute("title");
-    if (rel.indexOf("style") != -1 && title) {
-      links[i].disabled = (rel.indexOf("alt") != -1);
+
+  function set_preferred_stylesheet() {
+    var links = document.getElementsByTagName('link');
+    for (var i=0; i<links.length; i++) {
+      var rel = links[i].getAttribute('rel');
+      var title = links[i].getAttribute('title');
+      if (rel.indexOf('style') != -1 && title) {
+        links[i].disabled = (rel.indexOf('alt') != -1);
+      }
     }
   }
-}
-
-function get_preferred_stylesheet() {
-  var links = document.getElementsByTagName("link");
-  for (var i=0; i<links.length; i++) {
-    var rel = links[i].getAttribute("rel");
-    var title = links[i].getAttribute("title");
-    if (rel.indexOf("style") != -1 && rel.indexOf("alt") == -1 && title) {
-      return title;
+
+  function getPreferredStylesheet() {
+    var links = document.getElementsByTagName('link');
+    for (var i=0; i<links.length; i++) {
+      var rel = links[i].getAttribute('rel');
+      var title = links[i].getAttribute('title');
+      if (rel.indexOf('style') != -1 && rel.indexOf('alt') == -1 && title) {
+        return title;
+      }
     }
+    return null;
   }
-  return null;
-}
-
-function get_active_stylesheet() {
-  var links=document.getElementsByTagName("link");
-  for (var i=0; i<links.length; i++) {
-    var rel=links[i].getAttribute("rel");
-    var title=links[i].getAttribute("title");
-    if (rel.indexOf("style") != -1 && title && !links[i].disabled) {
-      return title;
+
+  function get_active_stylesheet() {
+    var links=document.getElementsByTagName('link');
+    for (var i=0; i<links.length; i++) {
+      var rel=links[i].getAttribute('rel');
+      var title=links[i].getAttribute('title');
+      if (rel.indexOf('style') != -1 && title && !links[i].disabled) {
+        return title;
+      }
     }
   }
-}
 
-window.onunload = function(e) {
-  var title = get_active_stylesheet();
-  $.cookie('style_cookie', title, 365);
-}
+  window.onunload = function(e) {
+    var title = get_active_stylesheet();
+    $.cookie('style_cookie', title, 365);
+  };
 
-window.onload = function(e) {
-  var cookie = $.cookie('style_cookie');
-  var title = cookie ? cookie : get_preferred_stylesheet();
-  set_stylesheet(title);
-}
+  window.onload = function(e) {
+    var cookie = $.cookie('style_cookie');
+    var title = cookie ? cookie : getPreferredStylesheet();
+    setStylesheet(title);
+  };
 
-$(document).ready(function() {
+  $(document).ready(function() {
 
-  $('.set-theme').click(function(e) {
-    set_stylesheet($(this).data('theme'));
-    return false;
-  });
+    var fortscene;
+    $('.set-theme').click(function(e) {
+      var theme = $(this).data('theme');
+      if (theme === '3d') {
+        var $container = $('#forttree-content-container');
+        if (typeof fortscene === 'undefined') {
+          var fortscene = new FortScene($container);
```

Comment:
'fortscene' is already defined.<br>'FortScene' is not defined.<br>'fortscene' is defined but never used.